numpy/core/tests/test_ufunc.py

```diff
@@ -2224,6 +2224,14 @@ def test_ufunc_at_advanced(self):
         np.maximum.at(a, [0], 0)
         assert_equal(a, np.array([1, 2, 3]))
 
+    def test_at_negative_indexes(self):
+        a = np.arange(10)
+        indxs = np.array([-1, 1, -1, 2])
+        np.add.at(a, indxs, 1)
+        assert a[-1] == 11  # issue 24147
+        assert a[1] == 2
+        assert a[2] == 3
+
```

Comment:
An bad merge (locally) brought me here, but then I noticed that this can modify the original index array:
```
assert indxs[0] == -1
```
would fail.  I think it shows that passing pointers or full offsets rather than indices makes more sense here (we need a temporary array anyway), but I have to think about how a hot-fix should look like.
Probably either need a local buffer here, or push the fixing into the indexed loop.

(Maybe local buffer is good, since that is what I would want to do with that hot-fix anyway in some form, even if it may make sense to push the buffer for that case slightly further up.)